internal/controller/accessrequest_controller.go

```diff
@@ -112,9 +109,18 @@ func (r *AccessRequestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	logger.Debug("Validating AccessRequest")
-	err = ar.Validate()
+	err = r.Validate(ctx, ar)
 	if err != nil {
-		logger.Info("Validation error: %s", err)
+		if _, ok := err.(*AccessRequestConflictError); ok {
```

Comment:
Might be able to use `errors.Is` instead.

Reply:
I tried but I can't. `errors.Is` is only possible when the error instance is equal or the same. For that to work, I would need to define error types as:
```golang
var MyErrorType = errors.New("some error")
```
problems are:
- this is a public variable that can be changes by any code.
- error message must be always the same

With error types as structs, the type assertion is necessary.

Reply:
My bad, I should have said `errors.As`. I think you could use it in a way that's similar to how `incompleteSettingsError` is used in Argo CD. It also has a configurable error message, but it supports `errors.As`.

```go
	var incompleteSettingsErr *incompleteSettingsError
	return errors.As(err, &incompleteSettingsErr)
```

Might not be worth the extra line to declare the error. If you look at it and still don't like it, feel free to resolve.

Reply:
sure. I just want to check the type in this case and not instantiate a new variable with the error. I prefer to keep it as is. Too bad golang doesn't provide a nicer API to validate error types.